cmd/containeragent/unit/agent.go

```diff
@@ -290,7 +290,9 @@ func (c *containerUnitAgent) workers() (worker.Worker, error) {
 	}
 	manifolds := Manifolds(cfg)
 
-	eng, err := dependency.NewEngine(engine.DependencyEngineConfig())
+	metrics := engine.NewMetrics()
+	workerMetricsSink := metrics.ForModel(agentConfig.Model())
+	eng, err := dependency.NewEngine(engine.DependencyEngineConfig(workerMetricsSink))
```

Comment:
It seems odd to inject this into DependencyEngineConfig vs it just being another dependency that gets added that workers can ask for.
It seems especially odd that containeragent/unit/agent.go is creating a new metrics object.

Reply:
The problem is that we can only register the one metric instance for Prometheus with the same measurements (gauges etc). The dependency engine doesn't know about models and nor should it, so we can't pass it when we do `RecordStart`. That only leaves registering one metric and capturing the model uuid at the time the worker starts.

I can pass in the model name at the same time as the model uuid, and I can potentially pass it in with the config, but the restrictions of the Prometheus registration is preventing a nicer implementation.